cns/restserver/restserver.go

```diff
@@ -2055,3 +2115,140 @@ func (service *HTTPRestService) SendNCSnapShotPeriodically(ncSnapshotIntervalInM
 		}
 	}
 }
+
+// isNCWaitingForUpdate :- Try to retrieve NC version on NMA
+func (service *HTTPRestService) isNCWaitingForUpdate(ncVersion, ncid string) (waitingForUpdate bool, returnCode int, message string) {
+	getNCVersionURL, ok := versionURLs.Load(ncid)
+	if !ok {
+		returnCode = NotFound
+		message = fmt.Sprintf("[Azure-CNS] Network container %s not found", ncid)
+		return
+	}
+
+	response, err := nmagentclient.GetNetworkContainerVersion(ncid, getNCVersionURL.(string))
+	if err == nil {
+		if response.StatusCode == http.StatusOK {
+			var versionResponse nmagentclient.NetworkContainerResponse
+			rBytes, _ := ioutil.ReadAll(response.Body)
+			json.Unmarshal(rBytes, &versionResponse)
+			if versionResponse.ResponseCode == "200" {
+				v2Str := versionResponse.Version
+				v1, _ := strconv.Atoi(ncVersion)
+				v2, _ := strconv.Atoi(v2Str)
+				if v1 > v2 {
+					waitingForUpdate = true
+					returnCode = NetworkContainerPendingStatePropagation
+					message = fmt.Sprintf("[Azure-CNS] Network container %s v%d had not propagated to respective NMA w/ v%d", ncid, v1, v2)
+				}
+			} else {
+				returnCode = UnexpectedError
+				message = fmt.Sprintf("[Azure-CNS] Failed to get NC version from response %s for NC %s", rBytes, ncid)
+			}
+		} else {
+			returnCode = UnexpectedError
+			message = fmt.Sprintf("[Azure-CNS] Failed to get NC version with http status %d", response.StatusCode)
+		}
+	}
+
+	return
```

Comment:
what do you mean?

Reply:
The default value for waitingForUpdate is false. If NMA api returns an error ( err != nil ), you just return on 2159.
This would assume that vfp programming is done, no?

Reply:
fixed, added err msg and returncode